FlowCrypt/Models/Realm Models/KeypairRealmObject.swift

```diff
@@ -61,17 +63,28 @@ extension KeyInfoRealmObject {
             throw KeyInfoError.missingPrimaryFingerprint
         }
 
+        self.primaryKey = primaryFingerprint + user.email
         self.primaryFingerprint = primaryFingerprint
         self.passphrase = passphrase
         self.source = source.rawValue
         self.user = user
     }
 }
 
-extension KeyInfoRealmObject {
+extension KeypairRealmObject {
     /// associated user email
     var account: String {
         guard let email = user?.email else { fatalError() }
         return email
     }
 }
+
+extension KeypairRealmObject {
+    static func createPrimaryKey(primaryFingerprint: String, email: String) -> String {
+        var hash = SHA256()
```

Comment:
@tomholub
Let's try to use hash function, like you suggested before. BTW since we switched to iOS 15 we could use new `CryptoKit` instead of `CommonCrypto`.

Reply:
Forgot a comma again. Else hashes may collide.

Reply:
Not sure about this because we update has with two values, not use hash function on concatenated string. Maybe we need a unit test to check this.

Reply:
Updating hash twice or concatenating the string first should amount to the same thing. You can update it a third time in between the two with a comma. Unit test is not needed as long as we do one thing and stick to it.

Reply:
```
let primaryFingerprint = "ABC"
let email = "test@example.com"

var h = SHA256()
h.update(data: primaryFingerprint.data(using: .utf8)!)
h.update(data: email.data(using: .utf8)!)
print(h.finalize())

var h1 = SHA256()
h1.update(data: (email + primaryFingerprint).data(using: .utf8)!)
print(h1.finalize())
```

output:
```
SHA256 digest: c709f433890955e1b8a818859b0c52b46acb414b2f069f80e3123e5270d5c548
SHA256 digest: 8fd18525b74f59bb0dd80376ec8a4d918e6e6e46f55c53f2ed2f35d18bca9e3e
```